plugins/services/src/js/components/modals/NewCreateServiceModalForm.js

```diff
@@ -367,28 +371,76 @@ class NewCreateServiceModalForm extends Component {
     });
   }
 
-  getSectionList() {
+  getFormDropdownList(navigationItems, activeTab, options = {}) {
+    const {isNested} = options;
+
+    return navigationItems.reduce((accumulator, item, index) => {
+      accumulator.push({
+        className: classNames({'page-header-menu-item-nested': isNested}),
+        id: item.id,
+        isActive: activeTab === item.id || (activeTab == null && index === 0),
+        label: item.label
+      });
+
+      if (item.children) {
+        accumulator = accumulator.concat(
+          this.getFormDropdownList(item.children, activeTab, {isNested: true})
+        );
+      }
+
+      return accumulator;
+    }, []);
+  }
+
+  getFormNavigationItems(appConfig, data) {
+    const serviceLabel = pluralize('Service', findNestedPropertyInObject(
+      appConfig,
+      'containers.length'
+    ) || 1);
+
+    const tabList = [
+      {
+        id: 'services',
+        label: serviceLabel,
+        children: this.getContainerList(data)
+      }
+    ];
+
     if (this.state.isPod) {
-      return [
-        <TabButton id="networking" label="Networking" key="multinetworking" />,
-        <TabButton id="volumes" label="Volumes" key="multivolumes" />,
-        <TabButton
-          id="healthChecks"
-          label="Health Checks"
-          key="multihealthChecks" />,
-        <TabButton
-          id="environment"
-          label="Environment"
-          key="multienvironment" />
-      ];
+      tabList.push(
```

Comment:
I decided to go against it here because this array is used for both the mobile view (which is a dropdown) and the sidebar menu itself. So these arrays can be transformed into one of the two desired structures.

Alternatively, I could construct the array for the dropdown from the composed components, if you think that'd be better. I'd be fine with either approach.